### ISSUE TITLE
Apple Silicon (M1) - backend won't start (dyld: missing symbol called) - After upgrading to 1.0.x

### ISSUE THREAD

**Issue by MrBlaise** [backstage/backstage#10476]
## Expected Behavior

When running `yarn start-backend` on a backstage app created by `npx @backstage/create-app` it should start up on an Apple Silicon (arm64) machine.

## Actual Behavior

The backend fails to start with the following message:
```
yarn run v1.22.17
$ yarn workspace backend start
$ backstage-cli package start
Build succeeded
dyld[89372]: missing symbol called
```

## Steps to Reproduce

1. Run `npx @backstage/create-app` (choose sqlite)
2. After the app is ready go to the root and run `yarn start-backend` (on an Apple Silicon machine)

## Context

It has worked previously on  `"version": "0.69.1"` (from backstage.json). Fails on `"version": "1.0.1"`

## Your Environment

OS: MacOs 12.2.1 (Apple Silicon M1 Max)

- Output of `yarn backstage-cli info`: 

```text
yarn backstage-cli info
yarn run v1.22.17
$ /Users/balazsrostas/bitrise/test-backstage/node_modules/.bin/backstage-cli info
OS:   Darwin 21.3.0 - darwin/arm64
node: v16.13.2
yarn: 1.22.17
cli:  0.16.0 (installed)

Dependencies:
  @backstage/app-defaults               1.0.0
  @backstage/backend-common             0.13.1
  @backstage/backend-tasks              0.2.1
  @backstage/catalog-client             1.0.0
  @backstage/catalog-model              1.0.0
  @backstage/cli-common                 0.1.8
  @backstage/cli                        0.16.0
  @backstage/config-loader              1.0.0
  @backstage/config                     1.0.0
  @backstage/core-app-api               1.0.0
  @backstage/core-components            0.9.2
  @backstage/core-plugin-api            1.0.0
  @backstage/errors                     1.0.0
  @backstage/integration-react          1.0.0
  @backstage/integration                1.0.0
  @backstage/plugin-api-docs            0.8.3
  @backstage/plugin-app-backend         0.3.30
  @backstage/plugin-auth-backend        0.12.3
  @backstage/plugin-auth-node           0.1.6
  @backstage/plugin-catalog-backend     1.0.0
  @backstage/plugin-catalog-common      1.0.0
  @backstage/plugin-catalog-graph       0.2.15
  @backstage/plugin-catalog-import      0.8.6
  @backstage/plugin-catalog-react       1.0.0
  @backstage/plugin-catalog             1.0.0
  @backstage/plugin-github-actions      0.5.3
  @backstage/plugin-org                 0.5.3
  @backstage/plugin-permission-common   0.5.3
  @backstage/plugin-permission-node     0.5.5
  @backstage/plugin-permission-react    0.3.4
  @backstage/plugin-proxy-backend       0.2.24
  @backstage/plugin-scaffolder-backend  1.0.0
  @backstage/plugin-scaffolder-common   1.0.0
  @backstage/plugin-scaffolder          1.0.0
  @backstage/plugin-search-backend-node 0.5.2
  @backstage/plugin-search-backend      0.4.8
  @backstage/plugin-search-common       0.3.2
  @backstage/plugin-search              0.7.4
  @backstage/plugin-tech-radar          0.5.10
  @backstage/plugin-techdocs-backend    1.0.0
  @backstage/plugin-techdocs-node       1.0.0
  @backstage/plugin-techdocs            1.0.0
  @backstage/plugin-user-settings       0.4.2
  @backstage/release-manifests          0.0.2
  @backstage/search-common              0.3.2
  @backstage/test-utils                 1.0.0
  @backstage/theme                      0.2.15
  @backstage/types                      1.0.0
  @backstage/version-bridge             1.0.0
✨  Done in 0.91s.
```

**Comment by MrBlaise** [comment]
Reinstalling nodejs using x86_64 architecture (using this guide for example: https://gist.github.com/adrienjoly/e29a6e73fb7e701eefd80ff9bde9abeb) is currently the workaround.

**Comment by goenning** [comment]
Same here, is it a new package that backstage is using which is incompatible with M1?

**Comment by benjdlambert** [comment]
Hmm so this is pretty weird. I got a new M1 yesterday, and it works fine using 14.x, but 16.x throws this. Can you both confirm that it works fine on 14.x? And maybe help out trying to work out what library it is that doesn't like 16?

**Comment by MrBlaise** [comment]
> Hmm so this is pretty weird. I got a new M1 yesterday, and it works fine using 14.x, but 16.x throws this. Can you both confirm that it works fine on 14.x? And maybe help out trying to work out what library it is that doesn't like 16?

When I install node 14.x using nvm it actually installs the x86 version. You can check that by running `node -p process.arch` when using the 14.x. That would explain why it works.

**Comment by benjdlambert** [comment]
Nice catch, so this is where it gets interesting. A new `create-app` throws this error, but the main repo works fine in `arm64`. Guessing `yarn.lock` installs later versions of some binary dependency, but not sure which one it is. Need to find all the packages that have bindings and add logs before and after requiring them to see where it bombs out and lock down the dependency somewhere to one that we know works in `arm64` I guess.

**Comment by debo** [comment]
Hello there, I just faced this very same problem, so for now the solution is to run node 14.x? Asking so I can unlock myself and move forward. Please let me know, thanks a lot for your time and help.

**Comment by MrBlaise** [comment]
> Hello there, I just faced this very same problem, so for now the solution is to run node 14.x? Asking so I can unlock myself and move forward. Please let me know, thanks a lot for your time and help.

You can install the intel version of node 16.x, that works too, guide here: https://gist.github.com/adrienjoly/e29a6e73fb7e701eefd80ff9bde9abeb

**Comment by Rugvip** [comment]
Ran into this in a separate setup, even though I used a compatible `node-gyp`, so disregard the previous comment.

I narrowed it down to a problem with `dockerode`, it seems to happen when trying to instantiate a `new Docker()` client. I'd expect the actual issue to be in one of the lower level transitive dependencies, so probably https://github.com/mscdex/ssh2/issues/1146 and https://github.com/mscdex/cpu-features/issues/7

**Comment by ugho49** [comment]
A newer version of `better-sqlite3` is available (`7.5.1`). 
If you bump to this newer version, the problem should be fixed !

**Comment by freben** [comment]
We believe this is fixed now - please reopen if the problem persists after the mentioned upgrades.
